src/pkgmt/versioneer.py

```diff
@@ -214,6 +215,149 @@ def add_changelog_new_dev_section(self, dev_version):
             print('No CHANGELOG.{rst,md} found, skipping changelog editing...')
 
 
+class VersionerNonSetup:
```

Comment:
I'm guessing everything in `VersionerNonSetup` (except the `__init__`) is the same as in `Versioner`? We can simplify this a lot if we write an [abstract class.](https://www.geeksforgeeks.org/abstract-classes-in-python/). The abstract class will contain all the base implementation and the concrete classes (`Versioner`, and `VersionerNonSetup`) will only implement what's particular to them. Here's some pseudocode (it'll probably wont run):

```python
import abc

class AbstractVersioner(abc.ABC):
  def __init__(self, project_root='.'):
    self.package_name = self.find_package()
    # implementation continues
    # (same code as the current Versioner.__init__ beginning in line 85
    self.PACKAGE = path_to_src / self.package_name
    ...
   
  @abc.abstractmethod
  def find_package(self):
       pass

   # all the other methods currently implemented in Versioner go here
```


Then for Versioner

```python
# make it a subclass of AbstractVersioner
class Versioner(AbstractVersioner):
  # and implement find_package
  def find_package(self):
    # lines 73-84 from Versioner.__init__
```

then the new one:

```python
# make it a subclass of AbstractVersioner
class VersionerNonSetup(AbstractVersioner):
  # and implement find_package
  def find_package(self):
     # the other implementation that does not need the setup.py
```